### PR TITLE
feat(shorebird_cli): `shorebird init`

### PR DESCRIPTION
**Pull request by felangel**
## Description

- `shorebird init`
  - creates a new `shorebird.yaml` if it doesn't already exist
  - will update the `pubspec.yaml` assets to contain the `shorebird.yaml`
- `shorebird publish`
  - will direct users to run `shorebird init` if shorebird has not been initialized yet.

## Type of Change

- [X] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/shorebird_cli/README.md [hunk 1]
```diff
@@ -41,11 +49,27 @@ shorebird logout
...
+
+```bash
+shorebird apps list
+```
+
+**Sample**
+
+```
+shorebird apps list
+shorebird-counter: v1.0.0 (patch #1)
```
Comment (eseidel) [inline packages/shorebird_cli/README.md]: my-counter is probably more representative. 🤷

File: packages/shorebird_cli/lib/src/shorebird_config_mixin.dart [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+mixin ShorebirdConfigMixin on ShorebirdCommand {
+  bool get hasShorebirdYaml => getShorebirdYaml() != null;
+
+  bool get hasPubspecYaml => getPubspecYaml() != null;
+
+  bool get isShorebirdInitialized {
+    return hasShorebirdYaml && pubspecContainsShoreBirdYaml;
+  }
+
+  bool get pubspecContainsShoreBirdYaml {
```
Comment (eseidel) [inline packages/shorebird_cli/lib/src/shorebird_config_mixin.dart]: Shorebird

**Review by eseidel — APPROVED** [review]
Thanks! 🎉

File: packages/shorebird_cli/lib/src/shorebird_config_mixin.dart (hunk 2 above)
Comment (felangel) [reply packages/shorebird_cli/lib/src/shorebird_config_mixin.dart]: Good catch, fixed ✅

**Pull request merged**